src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionTests.ConnectionLifecycle.cs

```diff
@@ -427,6 +427,34 @@ public async Task CanCancelStartingConnectionAfterNegotiate()
                 }
             }
 
+            [Fact]
+            public async Task CancellationTokenFromStartPassedToTransport()
+            {
+                using (StartVerifiableLog())
+                {
+                    var cts = new CancellationTokenSource();
+                    var httpHandler = new TestHttpMessageHandler();
+
+                    await WithConnectionAsync(
+                        CreateConnection(httpHandler,
+                        transport: new TestTransport(onTransportStart: () => {
+                            // Cancel the token when the transport is starting  which will fail the startTask.
+                            cts.Cancel();
+                            return Task.CompletedTask;
+                        })),
+                        async (connection) =>
+                        {
+                            // We aggregate failures that happen when we start the transport. The operation cancelled exception will
+                            // be an inner exception.
+                            var ex = await Assert.ThrowsAsync<AggregateException>(async () => await connection.StartAsync(cts.Token)).OrTimeout();
+                            Assert.Equal(3, ex.InnerExceptions.Count);
+                            var innerEx = ex.InnerExceptions[2];
+                            var innerInnerEx = innerEx.InnerException;
+                            Assert.IsType<OperationCanceledException>(innerInnerEx);
```

Comment:
This looks a little fragile. I'd probably be OK with just verifying `var ex = await Assert.ThrowsAnyAsync<Exception>(async () => await connection.StartAsync(cts.Token)).OrTimeout();`